### PR TITLE
Port can opening from /vg/

### PR DESCRIPTION
**Pull request by ghost**
:cl: More Robust Than You
add: Nanotrasen has added lids to their soda that POP! when opened. Up to 3 possible sounds!
/:cl:

### CODE REVIEW

**Comment by Niknakflak** [comment]
HAS SCIENCE GONE TOO FAR?

**Comment by PraiseRatvar** [comment]
yES better than organ update

**Comment by MoreRobustThanYou** [comment]
copy/paste is a magical thing

**Comment by Tacolizard** [comment]
have it make a fart sound when the clown opens it.

**Comment by MoreRobustThanYou** [comment]
@Tacolizard I'm not that great with code, how would i check if the user was a clown?

**Comment by DoctoreMad** [comment]
Use mob MIND

**Comment by DoctoreMad** [comment]
But don't do that, I go to hippie when I feel like I haven't had enough fart jokes.

**Comment by Supermichael777** [comment]
make it set spillable = true

**Comment by MoreRobustThanYou** [comment]
@Supermichael777 done

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 1]
```diff
@@ -312,6 +326,9 @@
 
 /obj/item/weapon/reagent_containers/food/drinks/soda_cans
 	name = "soda can"
+	flags = FALSE
+	spillable = 0
```
Comment (Cyberboss) [inline code/modules/food_and_drinks/drinks/drinks.dm]: FALSE

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 2]
```diff
@@ -312,6 +326,9 @@
...
+	var/list/open_sounds = list('sound/effects/can_open1.ogg', 'sound/effects/can_open2.ogg', 'sound/effects/can_open3.ogg')
```
Comment (Cyberboss) [inline code/modules/food_and_drinks/drinks/drinks.dm]: make this list static, and make the name more unique

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 3]
```diff
@@ -321,6 +338,16 @@
 		crushed_can.icon_state = icon_state
 		qdel(src)
 	..()
+	
+
+/obj/item/weapon/reagent_containers/food/drinks/soda_cans/attack_self(mob/user)
+	if(!is_open_container())
+		to_chat(user, "You pull back the tab of \the [src] with a satisfying pop.")
+		flags |= OPENCONTAINER
+		playsound(user, pick(open_sounds), 50, 1)
```
Comment (Cyberboss) [inline code/modules/food_and_drinks/drinks/drinks.dm]: playsound(src,...

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 4]
```diff
@@ -321,6 +338,16 @@
...
+		spillable = 1
```
Comment (Cyberboss) [inline code/modules/food_and_drinks/drinks/drinks.dm]: TRUE

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 5]
```diff
@@ -312,6 +326,9 @@
 
 /obj/item/weapon/reagent_containers/food/drinks/soda_cans
 	name = "soda can"
+	flags = FALSE
```
Comment (Cyberboss) [inline code/modules/food_and_drinks/drinks/drinks.dm]: 0

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/food_and_drinks/drinks/drinks.dm (hunk 1 above)
Comment (MoreRobustThanYou) [inline code/modules/food_and_drinks/drinks/drinks.dm]: the reason it was like that in /vg/ code, there was a comment saying "so admins can change the sounds"

File: code/modules/food_and_drinks/drinks/drinks.dm (hunk 2 above)
Comment (MoreRobustThanYou) [inline code/modules/food_and_drinks/drinks/drinks.dm]: the reason it was like that in /vg/ code, there was a comment saying "so admins can change the sounds"

**Comment by PKPenguin321** [comment]
We've deliberately removed features like these in the past though, such as monkey cube wrappers

**Comment by MoreRobustThanYou** [comment]
all of @Cyberboss's requests have been implemented, except for making that static (wouldn't that make the sounds unchangeable via var-editing?)

**Comment by Cyberboss** [comment]
No they'd still be var-editable. GLOB depends on that principle

**Comment by MoreRobustThanYou** [comment]
Ok, now ALL of Cyberboss's requests have been implemented.

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 6]
```diff
@@ -312,6 +326,9 @@
 
 /obj/item/weapon/reagent_containers/food/drinks/soda_cans
 	name = "soda can"
+	flags = 0
+	spillable = FALSE
+	var/static/list/can_pop_sounds = list('sound/effects/can_open1.ogg', 'sound/effects/can_open2.ogg', 'sound/effects/can_open3.ogg')
```
Comment (Cyberboss) [inline code/modules/food_and_drinks/drinks/drinks.dm]: Wait, sorry, this isn't how it should be done

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 7]
```diff
@@ -321,6 +338,16 @@
 		crushed_can.icon_state = icon_state
 		qdel(src)
 	..()
+	
+
+/obj/item/weapon/reagent_containers/food/drinks/soda_cans/attack_self(mob/user)
+	if(!is_open_container())
+		to_chat(user, "You pull back the tab of \the [src] with a satisfying pop.")
+		flags |= OPENCONTAINER
+		playsound(src, pick(open_sounds), 50, 1)
```
Comment (Cyberboss) [inline code/modules/food_and_drinks/drinks/drinks.dm]: Use playsound(src, "can_open", ... and create a "can_open" entry in get_sfx: https://github.com/tgstation/tgstation/blob/master/code/game/sound.dm#L101

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by MoreRobustThanYou** [comment]
Ok... I think I've done it right.

**Comment by MoreRobustThanYou** [comment]
@Cyberboss did I do it right?

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 8]
```diff
@@ -344,7 +343,7 @@
 	if(!is_open_container())
 		to_chat(user, "You pull back the tab of \the [src] with a satisfying pop.")
 		flags |= OPENCONTAINER
-		playsound(src, pick(open_sounds), 50, 1)
+		playsound(user.loc, "can_open", 50, 1)
```
Comment (Cyberboss) [inline code/modules/food_and_drinks/drinks/drinks.dm]: playsound(src,...

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by MoreRobustThanYou** [comment]
Ok, done.

**Review by Cyberboss — DISMISSED** [review]

**Comment by Kromgar** [comment]
@PKPenguin321 for good fucking reason. Monkey cube rappers were fucking tedious

**Comment by MoreRobustThanYou** [comment]
I tested locally and it didn't work...?

**Comment by MoreRobustThanYou** [comment]
It just allowed me to drink from Space Cola... and I couldn't open the tab at all. Just did "transfer rate now x"

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 9]
```diff
@@ -50,6 +54,10 @@
 /obj/item/weapon/reagent_containers/food/drinks/afterattack(obj/target, mob/user , proximity)
 	if(!proximity) return
 	if(istype(target, /obj/structure/reagent_dispensers)) //A dispenser. Transfer FROM it TO us.
+	
+		if (!is_open_container())
```
Comment (optimumtact) [inline code/modules/food_and_drinks/drinks/drinks.dm]: this is just all drinks and not soda's specifically, so now you have a soda specific message for all drinks

Comment (MoreRobustThanYou) [inline code/modules/food_and_drinks/drinks/drinks.dm]: However, non-sodas would automatically have the OPEN CONTAINER flag.

**Comment by MoreRobustThanYou** [comment]
REEEE THIS DOESN'T WORK

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 10]
```diff
@@ -312,7 +326,9 @@
 
 /obj/item/weapon/reagent_containers/food/drinks/soda_cans
 	name = "soda can"
-
+	flags = 0
```
Comment (optimumtact) [inline code/modules/food_and_drinks/drinks/drinks.dm]: container_type, not flag

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 11]
```diff
@@ -321,6 +337,16 @@
 		crushed_can.icon_state = icon_state
 		qdel(src)
 	..()
+	
+
+/obj/item/weapon/reagent_containers/food/drinks/soda_cans/attack_self(mob/user)
+	if(!is_open_container())
+		to_chat(user, "You pull back the tab of \the [src] with a satisfying pop.") //Ahhhhhhhh
+		flags |= OPENCONTAINER
```
Comment (optimumtact) [inline code/modules/food_and_drinks/drinks/drinks.dm]: container_type not flags

**Comment by MoreRobustThanYou** [comment]
SHOULD work now. Testing locally right now.

**Comment by MoreRobustThanYou** [comment]
@Cyberboss CAN CONFIRM, IT WORKS NOW!

**Review by Cyberboss — APPROVED** [review]

File: code/modules/food_and_drinks/drinks/drinks.dm (hunk 9 above)
Comment (octareenroon91) [inline code/modules/food_and_drinks/drinks/drinks.dm]: Could go with "is still sealed"

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 12]
```diff
@@ -50,6 +54,10 @@
 /obj/item/weapon/reagent_containers/food/drinks/afterattack(obj/target, mob/user , proximity)
 	if(!proximity) return
 	if(istype(target, /obj/structure/reagent_dispensers)) //A dispenser. Transfer FROM it TO us.
+	
+		if (!is_open_container())
+			to_chat(user, "<span class='warning'>[target]'s tab isn't open!</span>")
```
Comment (Cyberboss) [inline code/modules/food_and_drinks/drinks/drinks.dm]: Change this message to not be soda specific

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/food_and_drinks/drinks/drinks.dm (hunk 12 above)
Comment (MoreRobustThanYou) [inline code/modules/food_and_drinks/drinks/drinks.dm]: Ok, even though there should be no situation where that triggers for non-soda, barring badminnery/varediting

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**